# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpChannel.h
# Commit: 55c0d49b821c
# Full Hash: 55c0d49b821c3bbe25f4bda4c4e5cca9e725287c
# Author: Paul Bone <pbone@mozilla.com>
# Date: 2019-11-01 16:13:11
# Regressor Bug: 1588412
# File Overlap Count: 2
# Description:
#   Bug 1588412 - (part 3) Remove MayChangeProcess calls fron nsHttpChannel r=mayhemer
#   
#   Also add assertions that enforce that fission requires
#   DocumentChannel.
#   
# ==============================================================================

diff -r e0e3a3390f76 -r 55c0d49b821c netwerk/protocol/http/nsHttpChannel.h
--- a/netwerk/protocol/http/nsHttpChannel.h	Thu Oct 31 18:36:56 2019 +0000
+++ b/netwerk/protocol/http/nsHttpChannel.h	Thu Oct 31 18:13:50 2019 +0000
@@ -332,6 +332,11 @@
   void AsyncContinueProcessResponse();
   MOZ_MUST_USE nsresult ContinueProcessResponse1();
   MOZ_MUST_USE nsresult ContinueProcessResponse2(nsresult);
+
+ private:
+  void AssertNotDocumentChennel();
+
+ public:
   void UpdateCacheDisposition(bool aSuccessfulReval, bool aPartialContentUsed);
   MOZ_MUST_USE nsresult ContinueProcessResponse3(nsresult);
   MOZ_MUST_USE nsresult ContinueProcessResponse4(nsresult);
